theano/tensor/blas.py

```diff
@@ -304,14 +305,16 @@ def default_blas_ldflags():
                 ['-L%s' % l for l in blas_info['library_dirs']] +
                 ['-l%s' % l for l in blas_info['libraries']] +
                 [])
-            if GCC_compiler.try_flags(ret):
+            if GCC_compiler.try_flags(ret) and try_blas_flag():
+                ret += ['-lblas']
                 return ' '.join(ret)
             # Try to add the anaconda lib directory to runtime loading of lib.
             # This fix some case with Anaconda 2.3 on Linux.
             if "Anaconda" in sys.version and "linux" in sys.platform:
                 lib_path = os.path.join(sys.prefix, 'lib')
                 ret.append('-Wl,-rpath,' + lib_path)
-                if GCC_compiler.try_flags(ret):
+                if GCC_compiler.try_flags(ret) and try_blas_flag():
+                    ret += ['-lblas']
```

Comment:
remove this addition to the flags. We already found the flags we think should be used, don't modify them. Idem elsewhere.